src/objects/StatController.ts

```diff
@@ -19,7 +20,14 @@ export class StatController {
      * @returns The updated StatController instance.
      */
     add(stat: Statistic<any, any>) {
-        this.statistics.push(stat);
+        // If the statistic requires api methods which aren't available
+        // on the user's OpenRCT2 version,
+        // replace it with an unsupported statistic widget.
+        if (context.apiVersion < stat.minimumApiVersion) {
+            this.statistics.push(new UnsupportedStatistic(stat));
+        } else {
+            this.statistics.push(stat);
+        }
```

Comment:
Handling `UnsupportedStatistic`s like this also runs into the issue of having the original stat initialized with undefined subscriptions. As much as having the `minimumApiVersion` be handled in the background is great, I think perhaps it's better to have it on a statistic-by-statistic basis. Say, in the `timeSpentStatistic` method:

```ts
const minimumApiVersion = 70;
if (context.apiVersion < minimumApiVersion) {
    return new Statistic(...);
}
else {
    return new UnsupportedStatistic(...);
}
```

Reply:
I like what you're thinking, but I think I've got a more elegant refactor. My refactor moves to a factory function to create the statistic instead of directly calling the constructor, so to add no additional friction for statistic creation.